### ISSUE TITLE
Fatal crash in for each loop

### ISSUE THREAD

**Issue by sergeyfitis** [realm/realm-java#602]
Hi, I have issue similar to https://github.com/realm/realm-java/issues/484 and to https://github.com/realm/realm-java/issues/589 . I trying to run query in cycle, check if this record not exists, then create it. Before ```for/each cycle``` at the same thread and the same realm instance query works fine. But when we entered to cycle it crashed. I use ```realm library version 0.74.0, Android studio 1.0rc```.
```
Song model:
```
```
public class Song extends RealmObject {
    private String id;
    private String ownerId;
    private String artist;
    private String title;
    private int duration;
    private String url;
    private String lyrics_id;
    private int genreId;
    private String albumId;
    private int index;
    private boolean needToRemove;
**** and their getters and setters ****
}
```

```
Part of the code where it crash:
```
```
{
 List<Song> songs = lists[0];
 Realm workerRealm = getRealmInstanceForWorkerThread();
 String myId = PreferenceHelper.getUserId();
 // Mark all songs need to remove true
 RealmResults<Song> songRealmList = workerRealm.where(Song.class)
          .equalTo("ownerId", myId)
          .findAll();                                         // Works fine
 workerRealm.beginTransaction();
 for (Song song : songRealmList) {
    song.setNeedToRemove(true);         // Works fine
 }
 for (Song song : songs) {                    // Entering to cycle
   // Check if the record exist
  Song querySong = workerRealm.where(Song.class)
         .beginGroup()
         .equalTo("id", song.getId())
         .equalTo("ownerId", song.getOwnerId())
         .endGroup()
         .findFirst();                                 // And here it crash              

 if (querySong == null) {
   querySong = workerRealm.createObject(Song.class);
 }
 querySong.setOwnerId(song.getOwnerId());
 querySong.setUrl(song.getUrl());
 querySong.setLyrics_id(song.getLyrics_id());
 querySong.setGenreId(song.getGenreId());
 querySong.setDuration(song.getDuration());
 querySong.setTitle(song.getTitle());
 querySong.setId(song.getId());
 querySong.setAlbumId(song.getAlbumId());
 querySong.setArtist(song.getArtist());
 querySong.setNeedToRemove(song.isNeedToRemove());
 querySong.setIndex(song.getIndex());
}
```
Error message: ``` Method threw 'java.lang.RuntimeException' exception. Cannot evaluate io.realm.SongRealmProxy.toString() ```
  ```A/libc﹕ Fatal signal 11 (SIGSEGV), code 1, fault addr 0xa1e25ffc in tid -2049854000 (UpdateMultipleO)```

```UpdateMultipleO``` - it is my method where I do this( full name ```updateMultipleOwnSongsAndAlbumsInDB```).

**Comment by cmelchior** [comment]
Just a quick note that this has not been forgotten, but has been delayed by some needed changes to the core database. There is no timeframe, but we are actively working on it now.

**Comment by cmelchior** [comment]
Fixed in 0.89
